### ISSUE TITLE
[Bug] Metadata string value returned as datetime.date

### ISSUE THREAD

**Issue by mneira10** [pinecone-io/pinecone-python-client#161]
### Is this a new bug in the Pinecone Python client?

- [X] I believe this is a new bug in the Pinecone Python Client
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

If you upload a metadata field with a certain format, for example the string "1002.6.3" , pinecone will return the datetime object: `datetime.date(1002,6,2)` instead of the string `"1002.6.3"`

I'm not certain of all of the formats that lead to his behavior 

### Expected Behavior

The returned metadata value should be the **string** `"1002.6.3"` 

### Steps To Reproduce

1. upload a vector with metadata: `{"exampleField": "1002.6.3"}`
3. query the database for that vector. You'll see a `datetime.date` object instead of a string

### Environment

```markdown
- OS: MacOS Monterrey (12.2.1)
- Python: 3.11.2
- pinecone: 2.2.1
```

### Additional Context

Note that the value is correctly stored in pinecone but the sdk is returning it as a datetime object

**Comment by rajat08** [comment]
@mneira10 this is a known bug at the moment. This is because the openapi generated code we use for our REST client does this conversion as the spec does not clearly define all accepted openapi types. While we fix this, I suggest using GRPCIndex; it will be fine there.

**Comment by mneira10** [comment]
Thanks for the swift response! 

>I suggest using GRPCIndex; it will be fine there.

Good to know! Thanks 

---
As a possible workaround for others facing the same issue. You can adjust the formatting by adding a prefix or similar so that it won't be picked up as a date

**Comment by rajat08** [comment]
> 

Thanks for the suggestion :) 
I think it might be better to make these changes at the proto/spec level and avoid other data type-related issues. Our [v3 client](https://github.com/pinecone-io/pinecone-client) is in dev release atm (stable release should be out soon), but it is gRPC only and hence avoids some issues like these, which are downstream of our json-grpc transcoding decisions.

**Comment by masonchambersiii** [comment]
Was this ever resolved? Im still running into this issue?

**Comment by masonchambersiii** [comment]
Ended up just using the GRPC and it works fine -- but pretty surprised this issue hasnt been resolved for rest!

**Comment by chostyouwang** [comment]
Is there a follow-up repair plan, or do you strongly recommend using the gRPC method?

**Comment by chostyouwang** [comment]
> > Was this ever resolved? Im still running into this issue?
> 
> Not yet for REST. Can you use GRPCIndex?
How to use GRPCIndex? Do you have any reference demos?

**Comment by zeke-emerson** [comment]
@mneira10 This issue is fixed in v3.0.x of the Python client! To be clear, the fix allows the non-GRPC client to handle date strings as expected.

**Comment by mneira10** [comment]
>the fix allows the non-GRPC client to handle date strings as expected.

That's awesome! Thanks for clarifying that! 🚀
